### PR TITLE
Ports TG sleeping carp, the second.

### PR DESCRIPTION
**Pull request by hal9000PR**
## What Does This PR Do
original PR #15254

- Makes sleeping carp only deflect projectiles when throw intent is enabled.
- Makes sleeping carp users make a sound when shot.
- Makes projectiles shot at sleeping carp users reflect in random directions.
- Sleeping carp now costs 13tc.
- Removes the random stun on sleeping carps punches.
- Removes: wrist wrench, head kick, back kick, elbow drop (sadness), stomach knee.
- Adds these combos:  Keelhaul, Crashing Waves Kick, Gnashing Teeth.
- Carp are now friendly to sleeping carp users

Yet again thank you TG for code and ideas.

**KEELHAUL**

Steps: Harm, Grab.

Deals 40 stamina damage, 10 brute damage and stuns for 2-4 seconds.

**CRASHING WAVES KICK**

Steps: Harm, Disarm.

Deals 15 damage and sends the target 7 tiles away from you.

**GNASHING TEETH**

Steps: Harm, Harm.

Deals 20 damage and makes you shout like a lunatic.

## Why It's Good For The Game
Sleeping carp is used as a passive stat boost to become aid power gaming in a way to become unkillable. It is either used for Stacking a bunch of buffs and then becoming a murder machine with his grace/chainsaw/any other OP melee weapon or newer players who just get one tapped by a baton. Sleeping carp is not user friendly, its combos are extremely clunky and unwieldy and barely any body can remember them, this results to people not using the martial art for punching and kicking and instead for just hurr durr chainsaw. 

This PR makes sleeping carps combos more streamlined and easier to understand. One for stunning, one for ranged utility/disengage and one for damage. This improves the newer players experience with it, while also rewarding experienced players with toggling throw intent to deflect projectiles.

This implementation of sleeping carp is also much better for the transition away from instant stuns. Current sleeping carp is riddled with long stuns and weakens. This PR replaces these with shorter stuns and stamina damage. 

## Images of changes
103808254-9a855580-504f-11eb-9af0-5693975dbd64

## Changelog
:cl:
tweak: Makes sleeping carp only deflect projectiles when throw intent is enabled.
soundadd: Makes sleeping carp users make a sound when shot.
tweak: Makes projectiles shot at sleeping carp users reflect in random directions.
tweak: Sleeping carp now costs 13tc.
tweak: Removes the random stun on sleeping carps punches.
del: Removes wrist wrench, head kick, back kick, elbow drop and stomach knee.
add: Adds these combos:  Keelhaul deals light damage, 40 stamina damage and a brief stun , Crashing Waves Kick deals 15 damage and sends the target flying, Gnashing Teeth deals 20 damage and makes you shout like a lunatic.
/:cl:

### CODE REVIEW

**Comment by Wojz** [comment]
Charlie, this is the 7th week in a row you've shown martial arts in class.

**Comment by ppi13** [comment]
Honestly, my only issue with this is that carp users won't be randomly detected by throwing kisses around.

File: code/modules/martial_arts/combos/sleeping_carp/gnashing_teeth.dm [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/datum/martial_combo/sleeping_carp/gnashing_teeth
+	name = "Gnashing Teeth"
+	steps = list(MARTIAL_COMBO_STEP_HARM, MARTIAL_COMBO_STEP_HARM)
+	explaination_text = "Every second, concequtive, punch will deal extra damage and you will shout to send fear into your opponents heart"
```
Comment (SteelSlayer) [inline code/modules/martial_arts/combos/sleeping_carp/gnashing_teeth.dm:4]:
```suggestion
	explaination_text = "Every second, consecutive, punch will deal extra damage and you will shout to send fear into your opponents heart"
```

File: code/modules/martial_arts/combos/sleeping_carp/kneehaul.dm [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/datum/martial_combo/sleeping_carp/kneehaul
+	name = "Kneehaul"
+	steps = list(MARTIAL_COMBO_STEP_HARM, MARTIAL_COMBO_STEP_GRAB)
+	explaination_text = "Kick an opponent to the floor, knocking them down and dealing stamina damage to them! "
```
Comment (SteelSlayer) [inline code/modules/martial_arts/combos/sleeping_carp/kneehaul.dm:4]:
```suggestion
	explaination_text = "Kick an opponent to the floor, knocking them down and dealing stamina damage to them!"
```

File: code/modules/martial_arts/martial.dm [hunk 3]
```diff
@@ -6,19 +6,31 @@
...
+	/// If the martial art has it's own explaination verb
+	var/has_explaination_verb = FALSE
+
+	/// What combos can the user do? List of combo types
+	var/list/combos = list()
+	/// What combos are currently (possibly) being performed
+	var/list/datum/martial_art/current_combos = list()
+	/// When the last hit happened
+	var/last_hit = 0
+	// If the user is preparing a martial arts stance
```
Comment (SteelSlayer) [inline code/modules/martial_arts/martial.dm:32]: While you're at it
```suggestion
	/// If the user is preparing a martial arts stance
```

File: code/modules/martial_arts/martial.dm [hunk 4]
```diff
@@ -266,6 +278,7 @@
 
 	var/datum/martial_art/the_sleeping_carp/theSleepingCarp = new(null)
 	theSleepingCarp.teach(user)
+	user.faction |= list("carp") //makes space carp peaceful to sleeping carp users
```
Comment (SteelSlayer) [inline code/modules/martial_arts/martial.dm:281]: You're already doing this in `teach()`

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 5]
```diff
@@ -36,10 +36,16 @@ emp_act
 		P.on_hit(src, 100, def_zone)
 		return 2
 
-	if(mind?.martial_art?.deflection_chance) //Some martial arts users can deflect projectiles!
+	if(mind?.martial_art?.deflection_chance && in_throw_mode) //Some martial arts users can deflect projectiles!
```
Comment (SteelSlayer) [inline code/modules/mob/living/carbon/human/human_defense.dm:39]: What if we want some martial arts to reflect things while not in throw mode? Perhaps have a proc on the martial art called `try_deflect()` or whatever, that returns the `prob(deflection chance)` by default. And then carp would additionally return whether they were in throw mode.

Comment (hal9000PR) [reply code/modules/mob/living/carbon/human/human_defense.dm]: now you all martial arts deflect normally however sleeping carp only deflects on throw mode

File: code/modules/martial_arts/sleeping_carp.dm [hunk 6]
```diff
@@ -40,3 +40,8 @@
 
 /datum/martial_art/the_sleeping_carp/explaination_footer(user)
 	to_chat(user, "<b><i>In addition, by having your throw mode on when being shot at, you enter an active defense mode where you will block and deflect all projectiles fired at you!</i></b>")
+
+/datum/martial_art/the_sleeping_carp/try_deflect(mob/user)
+	if(user.in_throw_mode && prob(deflection_chance)) // in case an admin wants to var edit carp to have less deflection chance
+		return TRUE
+	return FALSE
```
Comment (SteelSlayer) [inline code/modules/martial_arts/sleeping_carp.dm:47]: You basically did what I had in mind, just some streamlining now though.

In the base proc you can shorten this too:
```dm
/datum/martial_art/proc/try_deflect(mob/user)
	return prob(deflection_chance)
```

And then here you can do:

```dm
/datum/martial_art/the_sleeping_carp/try_deflect(mob/user)
	return ..() && user.in_throw_mode // in case an admin wants to var edit carp to have less deflection chance
```
Which takes advantage of the parent proc.

File: code/modules/martial_arts/combos/sleeping_carp/gnashing_teeth.dm [hunk 7]
```diff
@@ -0,0 +1,18 @@
+/datum/martial_combo/sleeping_carp/gnashing_teeth
+	name = "Gnashing Teeth"
+	steps = list(MARTIAL_COMBO_STEP_HARM, MARTIAL_COMBO_STEP_HARM)
+	explaination_text = "Every second, consecutive, punch will deal extra damage and you will shout to send fear into your opponents heart"
```
Comment (S34NW) [inline code/modules/martial_arts/combos/sleeping_carp/gnashing_teeth.dm:4]:
```suggestion
	explaination_text = "Every second consecutive punch deals extra damage, and you will shout to strike fear into your opponent's heart."
```

File: code/modules/martial_arts/combos/sleeping_carp/keehaul.dm [hunk 8]
```diff
@@ -0,0 +1,15 @@
+/datum/martial_combo/sleeping_carp/keehaul
+	name = "Keehaul"
+	steps = list(MARTIAL_COMBO_STEP_HARM, MARTIAL_COMBO_STEP_GRAB)
+	explaination_text = "Kick an opponent to the floor, knocking them down and dealing stamina damage to them!"
```
Comment (S34NW) [inline code/modules/martial_arts/combos/sleeping_carp/keehaul.dm:4]:
```suggestion
	explaination_text = "Kicks an opponent to the floor, knocking them down and dealing stamina damage to them!"
```

**Review by S34NW — COMMENTED** [review]
Just some grammar things there and there, everything else looks fine.

**Review by Kyep — APPROVED** [review]

File: code/datums/uplink_item.dm [hunk 9]
```diff
@@ -849,11 +849,11 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 /datum/uplink_item/stealthy_weapons/martialarts
 	name = "Martial Arts Scroll"
 	desc = "This scroll contains the secrets of an ancient martial arts technique. You will master unarmed combat, \
-			deflecting all ranged weapon fire, but you also refuse to use dishonorable ranged weaponry. \
+			deflecting ranged weapon fire when we are in a defensive stance, but you also refuse to use dishonorable ranged weaponry. \
```
Comment (S34NW) [inline code/datums/uplink_item.dm:852]:
```suggestion
			deflecting ranged weapon fire when you are in a defensive stance (throw mode), but you will also refuse to use dishonorable ranged weaponry. \
```

Comment (S34NW) [reply code/datums/uplink_item.dm:852]: perhaps it would be better to make this clear from an OOC perspective?

File: code/modules/martial_arts/sleeping_carp.dm [hunk 10]
```diff
@@ -19,17 +20,30 @@
...
+					Your hand-to-hand combat has become much more effective, and you are now able to deflect any projectiles directed toward you when in throw mode. \
+					However, you are also unable to use any ranged weaponry. \
+					You can learn more about your newfound art by using the Recall Teachings verb in the Sleeping Carp tab.</span>")
+
+/datum/martial_art/the_sleeping_carp/remove(mob/living/carbon/human/H)
+	. = ..()
+	H.faction -= "carp"// :C
+
+/datum/martial_art/the_sleeping_carp/explaination_footer(user)
+	to_chat(user, "<b><i>In addition, by having your throw mode on when being shot at, you enter an active defense mode where you will block and deflect all projectiles fired at you!</i></b>")
```
Comment (S34NW) [inline code/modules/martial_arts/sleeping_carp.dm:46]:
```suggestion
	to_chat(user, "<b><i>In addition, by having your throw mode on when being shot at, you enter an active defensive mode where you will block and deflect all projectiles fired at you!</i></b>")
```

**Review by S34NW — COMMENTED** [review]
Some minor edits, there is also a typo in your CL:  `Keehaul` instead of  `Keelhaul`

File: code/datums/uplink_item.dm (hunk 9 above)
Comment (hal9000PR) [reply code/datums/uplink_item.dm:852]: Good idea

File: code/modules/martial_arts/combos/sleeping_carp/crashing_kick.dm [hunk 11]
```diff
@@ -0,0 +1,17 @@
+/datum/martial_combo/sleeping_carp/crashing_kick
+	name = "Crashing Waves Kick"
+	steps = list(MARTIAL_COMBO_STEP_HARM, MARTIAL_COMBO_STEP_DISARM)
+	explaination_text = "Kicks the target square in the chest, sending them flying."
+
+/datum/martial_combo/sleeping_carp/crashing_kick/perform_combo(mob/living/carbon/human/user, mob/living/target, datum/martial_art/MA)
+	if(target != user) // no you cannot kick yourself across rooms
+		user.do_attack_animation(target, ATTACK_EFFECT_KICK)
+		target.visible_message("<span class='warning'>[user] kicks [target] square in the chest, sending them flying!</span>",
+					"<span class='userdanger'>You are kicked square in the chest by [user], sending you flying!</span>",)
```
Comment (dearmochi) [inline code/modules/martial_arts/combos/sleeping_carp/crashing_kick.dm:10]:
```suggestion
					"<span class='userdanger'>You are kicked square in the chest by [user], sending you flying!</span>")
```

File: code/modules/martial_arts/combos/sleeping_carp/crashing_kick.dm [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+		playsound(get_turf(target), 'sound/effects/hit_kick.ogg', 50, TRUE, -1)
```
Comment (dearmochi) [inline code/modules/martial_arts/combos/sleeping_carp/crashing_kick.dm:11]:
```suggestion
		playsound(target, 'sound/effects/hit_kick.ogg', 50, TRUE, -1)
```

File: code/modules/martial_arts/combos/sleeping_carp/keehaul.dm [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+	explaination_text = "Kicks an opponent to the floor, knocking them down and dealing stamina damage to them!"
+
+/datum/martial_combo/sleeping_carp/keelhaul/perform_combo(mob/living/carbon/human/user, mob/living/target, datum/martial_art/MA)
+	user.do_attack_animation(target, ATTACK_EFFECT_KICK)
+	playsound(get_turf(target), 'sound/effects/hit_kick.ogg', 50, TRUE, -1)
+	target.apply_damage(10, BRUTE, BODY_ZONE_HEAD)
+	target.apply_damage(40, STAMINA)
+	target.Weaken(2)
+	target.visible_message("<span class='warning'>[user] kicks [target] in the head, sending them face first into the floor!</span>",
+						"<span class='userdanger'>You are kicked in the head by [user], sending you crashing to the floor!</span>",)
```
Comment (dearmochi) [inline code/modules/martial_arts/combos/sleeping_carp/keehaul.dm:13]:
```suggestion
						"<span class='userdanger'>You are kicked in the head by [user], sending you crashing to the floor!</span>")
```

**Review by dearmochi — APPROVED** [review]

File: code/modules/martial_arts/combos/sleeping_carp/gnashing_teeth.dm [hunk 14]
```diff
@@ -0,0 +1,18 @@
+/datum/martial_combo/sleeping_carp/gnashing_teeth
+	name = "Gnashing Teeth"
+	steps = list(MARTIAL_COMBO_STEP_HARM, MARTIAL_COMBO_STEP_HARM)
+	explaination_text = "Every second consecutive punch deals extra damage, and you will shout to strike fear into your opponent's heart."
+
+/datum/martial_combo/sleeping_carp/gnashing_teeth/perform_combo(mob/living/carbon/human/user, mob/living/target, datum/martial_art/MA)
+	user.do_attack_animation(target, ATTACK_EFFECT_PUNCH)
+	var/atk_verb = pick("precisely kick", "brutally chop", "cleanly hit", "viciously slam")
+	target.visible_message("<span class='danger'>[user] [atk_verb]s [target]!</span>",
+					"<span class='userdanger'>[user] [atk_verb]s you!</span>")
```
Comment (SabreML) [inline code/modules/martial_arts/combos/sleeping_carp/gnashing_teeth.dm:10]: Nitpick, but this might work better?
```suggestion
	var/atk_verb = pick("precisely kicks", "brutally chops", "cleanly hits", "viciously slams")
	target.visible_message("<span class='danger'>[user] [atk_verb] [target]!</span>",
					"<span class='userdanger'>[user] [atk_verb] you!</span>")
```

File: code/modules/martial_arts/combos/sleeping_carp/gnashing_teeth.dm [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+	playsound(get_turf(target), 'sound/weapons/punch1.ogg', 35, TRUE, -1)
+	add_attack_logs(user, target, "Melee attacked with martial-art [MA] :  Gnashing Teeth", ATKLOG_ALL)
```
Comment (SabreML) [inline code/modules/martial_arts/combos/sleeping_carp/gnashing_teeth.dm:12]: Got two spaces here
```suggestion
	add_attack_logs(user, target, "Melee attacked with martial-art [MA] : Gnashing Teeth", ATKLOG_ALL)
```

File: code/modules/martial_arts/combos/sleeping_carp/gnashing_teeth.dm [hunk 16]
```diff
@@ -0,0 +1,18 @@
...
+	target.apply_damage(20, BRUTE)
+	if(target.health >= 0)
+		user.say(pick("HUAH!", "HYA!", "CHOO!", "WUO!", "KYA!", "HUH!", "HIYOH!", "CARP STRIKE!", "CARP BITE!"))
+	else
+		user.say(pick("BANZAIII!", "KIYAAAA!", "OMAE WA MOU SHINDEIRU!", "YOU CAN'T SEE ME!", "MY TIME IS NOW!", "COWABUNGA")) // COWABUNGA
```
Comment (SabreML) [inline code/modules/martial_arts/combos/sleeping_carp/gnashing_teeth.dm:17]: Why the health threshold?

Comment (SabreML) [inline code/modules/martial_arts/combos/sleeping_carp/gnashing_teeth.dm:17]: Also,
```suggestion
		user.say(pick("BANZAIII!", "KIYAAAA!", "OMAE WA MOU SHINDEIRU!", "YOU CAN'T SEE ME!", "MY TIME IS NOW!", "COWABUNGA!")) // COWABUNGA
```

File: code/modules/martial_arts/combos/sleeping_carp/keehaul.dm [hunk 17]
```diff
@@ -0,0 +1,15 @@
+/datum/martial_combo/sleeping_carp/keelhaul
+	name = "Keelhaul"
```
Comment (SabreML) [inline code/modules/martial_arts/combos/sleeping_carp/keehaul.dm:2]: Typo in the file name:
`code/modules/martial_arts/combos/sleeping_carp/keehaul.dm`
vs
`code/modules/martial_arts/combos/sleeping_carp/keelhaul.dm`

File: code/modules/martial_arts/combos/sleeping_carp/keehaul.dm [hunk 18]
```diff
@@ -0,0 +1,15 @@
...
+/datum/martial_combo/sleeping_carp/keelhaul/perform_combo(mob/living/carbon/human/user, mob/living/target, datum/martial_art/MA)
+	user.do_attack_animation(target, ATTACK_EFFECT_KICK)
+	playsound(get_turf(target), 'sound/effects/hit_kick.ogg', 50, TRUE, -1)
+	target.apply_damage(10, BRUTE, BODY_ZONE_HEAD)
+	target.apply_damage(40, STAMINA)
+	target.Weaken(2)
+	target.visible_message("<span class='warning'>[user] kicks [target] in the head, sending them face first into the floor!</span>",
+						"<span class='userdanger'>You are kicked in the head by [user], sending you crashing to the floor!</span>")
+	add_attack_logs(user, target, "Melee attacked with martial-art [MA] : Keelhaul", ATKLOG_ALL)
+	return MARTIAL_COMBO_DONE
```
Comment (SabreML) [inline code/modules/martial_arts/combos/sleeping_carp/keehaul.dm:15]: This should probably have a check to see if the target is on the floor already, and do something different if they are.
`"You are kicked in the head by [user], sending you crashing to the floor!"` doesn't make sense really if you're lying down already.

File: code/modules/martial_arts/martial.dm [hunk 19]
```diff
@@ -6,19 +6,31 @@
...
-	var/deflection_chance = 0 //Chance to deflect projectiles
-	var/block_chance = 0 //Chance to block melee attacks using items while on throw mode.
+	/// The permanent style
+	var/datum/martial_art/base = null
+	/// Chance to deflect projectiles while on throw mode
+	var/deflection_chance = 0
+	/// Can it reflect projectiles in a random direction?
+	var/reroute_deflection = FALSE
+	///Chance to block melee attacks using items while on throw mode.
+	var/block_chance = 0
```
Comment (SabreML) [inline code/modules/martial_arts/martial.dm:16]: I'm not actually sure what the standard is for punctuation in comments, but some of these have full stops and some don't. Personally I'd just add it on to all of them.

File: code/modules/martial_arts/martial.dm [hunk 20]
```diff
@@ -6,19 +6,31 @@
...
-	var/has_explaination_verb = FALSE	// If the martial art has it's own explaination verb
-
-	var/list/combos = list()							// What combos can the user do? List of combo types
-	var/list/datum/martial_art/current_combos = list()	// What combos are currently (possibly) being performed
-	var/last_hit = 0									// When the last hit happened
-	var/in_stance = FALSE // If the user is preparing a martial arts stance
+	///set to TRUE to prevent users of this style from using guns (sleeping carp, highlander). They can still pick them up, but not fire them.
+	var/no_guns = FALSE
+	///message to tell the style user if they try and use a gun while no_guns = TRUE (DISHONORABRU!)
+	var/no_guns_message = ""
```
Comment (SabreML) [inline code/modules/martial_arts/martial.dm:21]: May as well for consistency with the rest:
```suggestion
	/// Set to TRUE to prevent users of this style from using guns (sleeping carp, highlander). They can still pick them up, but not fire them.
	var/no_guns = FALSE
	/// Message to tell the style user if they try and use a gun while no_guns = TRUE (DISHONORABRU!)
	var/no_guns_message = ""
```

File: code/modules/martial_arts/sleeping_carp.dm [hunk 21]
```diff
@@ -19,17 +20,30 @@
 	MARTIAL_ARTS_ACT_CHECK
 	A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
 	var/atk_verb = pick("punches", "kicks", "chops", "hits", "slams")
-	D.visible_message("<span class='danger'>[A] [atk_verb] [D]!</span>", \
+	D.visible_message("<span class='danger'>[A] [atk_verb] [D]!</span>",
 					  "<span class='userdanger'>[A] [atk_verb] you!</span>")
 	D.apply_damage(rand(10,15), BRUTE)
```
Comment (SabreML) [inline code/modules/martial_arts/sleeping_carp.dm:25]: This was here already, but:
```suggestion
	D.apply_damage(rand(10, 15), BRUTE)
```

File: code/modules/martial_arts/sleeping_carp.dm [hunk 22]
```diff
@@ -19,17 +20,30 @@
...
+
+/datum/martial_art/the_sleeping_carp/remove(mob/living/carbon/human/H)
+	. = ..()
+	H.faction -= "carp"// :C
+
+/datum/martial_art/the_sleeping_carp/explaination_footer(user)
+	to_chat(user, "<b><i>In addition, by having your throw mode on when being shot at, you enter an active defensive mode where you will block and deflect all projectiles fired at you!</i></b>")
+
+/datum/martial_art/the_sleeping_carp/try_deflect(mob/user)
+	return ..() && user.in_throw_mode // in case an admin wants to var edit carp to have less deflection chance
```
Comment (SabreML) [inline code/modules/martial_arts/sleeping_carp.dm:49]: For the sake of avoiding a couple of proc calls:
```suggestion
/datum/martial_art/the_sleeping_carp/try_deflect(mob/user)
	return user.in_throw_mode && ..() // in case an admin wants to var edit carp to have less deflection chance
```

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 23]
```diff
@@ -37,10 +37,16 @@ emp_act
 		return 2
 
 	if(mind?.martial_art?.deflection_chance) //Some martial arts users can deflect projectiles!
-		if(!lying && !HAS_TRAIT(src, TRAIT_HULK) && prob(mind.martial_art.deflection_chance)) //But only if they're not lying down, and hulks can't do it
+		if(!lying && !HAS_TRAIT(src, TRAIT_HULK) && mind?.martial_art?.try_deflect(src)) //But only if they're not lying down, and hulks can't do it
```
Comment (SabreML) [inline code/modules/mob/living/carbon/human/human_defense.dm:40]: You shouldn't need the extra checks here, since they're already done on the line above.
```suggestion
		if(!lying && !HAS_TRAIT(src, TRAIT_HULK) && mind.martial_art.try_deflect(src)) //But only if they're not lying down, and hulks can't do it
```

**Review by variableundefined — CHANGES_REQUESTED** [review]
What sabre requested please

File: code/modules/martial_arts/combos/sleeping_carp/gnashing_teeth.dm (hunk 16 above)
Comment (hal9000PR) [reply code/modules/martial_arts/combos/sleeping_carp/gnashing_teeth.dm:17]: if they are in crit they will shout the phrases that used to be shouted when elbow drop happened. so those are the "finisher" phrases

**Comment by hal9000PR** [comment]
oops- wait one second

File: code/modules/martial_arts/combos/sleeping_carp/gnashing_teeth.dm [hunk 24]
```diff
@@ -0,0 +1,18 @@
+/datum/martial_combo/sleeping_carp/gnashing_teeth
+	name = "Gnashing Teeth"
+	steps = list(MARTIAL_COMBO_STEP_HARM, MARTIAL_COMBO_STEP_HARM)
+	explaination_text = "Every second consecutive punch deals extra damage, and you will shout to strike fear into your opponent's heart."
```
Comment (SabreML) [inline code/modules/martial_arts/combos/sleeping_carp/gnashing_teeth.dm:4]: Is this a leftover from something? It seems to just do a fixed amount of damage now.

File: code/modules/martial_arts/martial.dm [hunk 25]
```diff
@@ -6,19 +6,31 @@
...
+
+	/// If the martial art has it's own explaination verb.
+	var/has_explaination_verb = FALSE
+
+	/// What combos can the user do? List of combo types.
+	var/list/combos = list()
+	/// What combos are currently (possibly) being performed.
+	var/list/datum/martial_art/current_combos = list()
+	/// When the last hit happened
+	var/last_hit = 0
```
Comment (SabreML) [inline code/modules/martial_arts/martial.dm:31]: Missed a spot!
```suggestion
	/// When the last hit happened.
	var/last_hit = 0
```

**Review by SabreML — COMMENTED** [review]
Last couple of things

**Review by SabreML — APPROVED** [review]

File: code/modules/martial_arts/combos/sleeping_carp/gnashing_teeth.dm [hunk 26]
```diff
@@ -0,0 +1,18 @@
...
+
+/datum/martial_combo/sleeping_carp/gnashing_teeth/perform_combo(mob/living/carbon/human/user, mob/living/target, datum/martial_art/MA)
+	user.do_attack_animation(target, ATTACK_EFFECT_PUNCH)
+	var/atk_verb = pick("precisely kicks", "brutally chops", "cleanly hits", "viciously slams")
+	target.visible_message("<span class='danger'>[user] [atk_verb] [target]!</span>",
+					"<span class='userdanger'>[user] [atk_verb] you!</span>")
+	playsound(get_turf(target), 'sound/weapons/punch1.ogg', 35, TRUE, -1)
```
Comment (Fox-McCloud) [inline code/modules/martial_arts/combos/sleeping_carp/gnashing_teeth.dm:11]: `25` instead of `35`.

File: code/modules/martial_arts/sleeping_carp.dm [hunk 27]
```diff
@@ -19,17 +20,30 @@
...
+	D.visible_message("<span class='danger'>[A] [atk_verb] [D]!</span>",
 					  "<span class='userdanger'>[A] [atk_verb] you!</span>")
-	D.apply_damage(rand(10,15), BRUTE)
-	playsound(get_turf(D), 'sound/weapons/punch1.ogg', 25, 1, -1)
-	if(prob(50))
-		A.say(pick("HUAH!", "HYA!", "CHOO!", "WUO!", "KYA!", "HUH!", "HIYOH!", "CARP STRIKE!", "CARP BITE!"))
-	if(prob(D.getBruteLoss()) && !D.lying)
-		D.visible_message("<span class='warning'>[D] stumbles and falls!</span>", "<span class='userdanger'>The blow sends you to the ground!</span>")
-		D.Weaken(4)
+	D.apply_damage(rand(10, 15), BRUTE)
```
Comment (Fox-McCloud) [inline code/modules/martial_arts/sleeping_carp.dm:25]: should probably get the impacted zone you're targeting, for consistency with other unarmed attacks.

File: code/modules/martial_arts/combos/sleeping_carp/keelhaul.dm [hunk 28]
```diff
@@ -0,0 +1,19 @@
...
+	target.apply_damage(40, STAMINA)
+	if(!target.IsWeakened() && !target.resting && !target.stat)
+		target.visible_message("<span class='warning'>[user] kicks [target] in the head, sending them face first into the floor!</span>",
+						"<span class='userdanger'>You are kicked in the head by [user], sending you crashing to the floor!</span>")
+	else
+		target.visible_message("<span class='warning'>[user] kicks [target] in the head, leaving them reeling in pain!</span>",
+							"<span class='userdanger'>You are kicked in the head by [user], and you reel in pain!</span>")
+	target.Weaken(2)
+	add_attack_logs(user, target, "Melee attacked with martial-art [MA] : Keelhaul", ATKLOG_ALL)
+	return MARTIAL_COMBO_DONE
```
Comment (Fox-McCloud) [inline code/modules/martial_arts/combos/sleeping_carp/keelhaul.dm:19]: Way want to re-look at this one, for consistency with TG---or look at what they had, pre-combat change, for ideas, as well.

**Comment by Fox-McCloud** [comment]
Just a few things, otherwise 👍

File: code/modules/martial_arts/combos/sleeping_carp/keelhaul.dm (hunk 28 above)
Comment (hal9000PR) [reply code/modules/martial_arts/combos/sleeping_carp/keelhaul.dm:19]: Right now, on TG. this knocks down for 4 seconds and deals 40 stamina damage if they are stood up. If they aren't stood up it disarms them and applies 40 stamina damage.  gitblame lists these changes as coming from a year ago. A short weaken and stamina damage seems to be the best I can do to replicate this? Maybe applying a slowdown and stamina damage, and then weakening if they are already slowed? I am not sure.
